webapp/src/pages/newEvent/newEvent.js

```diff
@@ -0,0 +1,805 @@
+import React, { Component } from "react";
```

Comment:
I'm not a fan of having separate components for editing and creating a new event - the reason being that it makes maintenance more difficult. For example, imagine in the future someone adds a new field to an event and updates the edit form but forgets to or doesn't realise that they also need to update the create new form (I've seen similar happen many times). It would be much better if you can integrate the "new" logic into the EventConfigComponent.

Reply:
Hi, Okay I understand the logic here but Im puzzled as to how eventConfig will be accessible in create mode when there is not event or event.key for that matter present?

Reply:
Hi Jean, you can have multiple routes pointing to the same component. So keep your route in App.js but point it to EventConfig instead of newEvent and then you already have some logic in event config that checks if `this.props.event` is populated, which is how you know if you're in create mode or edit mode.